### PR TITLE
feat: add auto-fix for MaintainerDeprecated rule

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

Adds auto-fix support for the BuildKit `MaintainerDeprecated` rule, which automatically replaces the deprecated `MAINTAINER` instruction with the OCI standard `LABEL org.opencontainers.image.authors`.

### Implementation
- **New enricher**: `internal/rules/buildkit/fixes/maintainer_deprecated.go`
  - Replaces `MAINTAINER` with `LABEL org.opencontainers.image.authors`
  - Handles various formats: simple email, name with email, quoted values
  - Safety level: `FixSafe` (formatting-only change)
- **Registration**: Added to enricher switch in `enricher.go`

### Testing
- **Unit tests**: Added comprehensive tests for various maintainer formats in `fixes_test.go`
- **Integration tests**:
  - Created test fixture in `testdata/maintainer-deprecated/`
  - Added detection test case to `TestCheck`
  - Added fix test case to `TestFix`
  - Updated existing snapshots to include the new fix

### Documentation
- Updated `RULES.md` to add 🔧 emoji indicating fix support

## Example Transformation

**Before:**
```dockerfile
MAINTAINER John Doe <john@example.com>
```

**After:**
```dockerfile
LABEL org.opencontainers.image.authors="John Doe <john@example.com>"
```

## Testing

All tests pass:
- ✅ Unit tests (fixes package)
- ✅ Integration tests
- ✅ golangci-lint
- ✅ PMD CPD (copy/paste detection)

## Test Plan

```bash
# Create test Dockerfile
echo 'FROM alpine:3.18
MAINTAINER John Doe <john@example.com>
RUN echo hello' > test.dockerfile

# Apply fix
go run . check --fix test.dockerfile

# Verify result
cat test.dockerfile
# Should show: LABEL org.opencontainers.image.authors="John Doe <john@example.com>"
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Violation reports now include automated suggested fixes for deprecated MAINTAINER instructions, proposing replacement with OCI-compliant LABEL org.opencontainers.image.authors entries.
  * Build/lint status labels updated to show a small inline wrench emoji alongside "Captured" status.

* **Tests**
  * Added integration and unit tests covering maintainer deprecation detection and fix application across many input formats and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: internal/rules/buildkit/fixes/maintainer_deprecated.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	if maintainerValue == "" {
+		return
+	}
+
+	// Remove surrounding quotes if present (they'll be re-added in the label)
+	maintainerValue = strings.Trim(maintainerValue, `"'`)
+
+	// Create the replacement LABEL instruction
+	newText := `LABEL org.opencontainers.image.authors="` + maintainerValue + `"`
+
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/buildkit/fixes/maintainer_deprecated.go]: ✅ Addressed in a2d9d64. Thanks @coderabbitai\!
